src/option.bif

```diff
@@ -10,6 +10,11 @@ module Option;
 static bool call_option_handlers_and_set_value(zeek::StringVal* name, const zeek::detail::IDPtr& i,
                                                zeek::ValPtr val, zeek::StringVal* location)
 	{
+	// when shutting down, don't call back into script land handlers as
+	// these may use resources that have already been shutdown.
+	if ( zeek::run_state::terminating )
```

Comment:
I'm a bit wondering now whether short-circuiting at such a low-level is quite right.

For the problem reported in #2010 specific to interaction with the config framework delivering events into script-land, we could also short-circuit in `InputConfig::new_value` when terminating and leave the `Option::set()` bif as is.

https://github.com/zeek/zeek/blob/master/scripts/base/frameworks/config/input.zeek#L53-L59

Thoughts?

Reply:
I'm fine with leaving it at this level. Moving it up a level into `InputConfig` potentially leaves the same bug to crop up in some other module at a later time.